.github/workflows/icu_valgrind.yml

```
@@ -0,0 +1,94 @@
+# for Valgrind
+
+name: GHA CI Valgrind
+
+on:
+  push:
+    branches:
+      - main
+      - 'maint/maint*'
+  pull_request:
+    branches: '**'
+
+jobs:
+  clang-valgrind-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install valgrind
+        run: |
+          set -ex;
+          sudo apt-get -y update;
+          sudo apt-get install -y valgrind;
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build with debug enable
+        env:
+          CC: clang
+          CXX: clang++
+        run: |
+          cd icu4c/source;
+          ./runConfigureICU --enable-debug Linux --disable-renaming && make -j4 tests;
+      - name: Test iotest with valgrind
+        run: |
+          cd icu4c/source/test/iotest
+          LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH \
+              valgrind --tool=memcheck --error-exitcode=1 --leak-check=full \
+              --show-reachable=yes ./iotest
+      - name: Test cintltst with valgrind
+        run: |
+          cd icu4c/source/test/cintltst
+          LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH \
+              valgrind --tool=memcheck --error-exitcode=1 --leak-check=full \
+              --show-reachable=yes ./cintltst
+      - name: Test icuinfo with valgrind
+        run: |
+          cd icu4c/source/tools/icuinfo
+          LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH \
+              valgrind --tool=memcheck --error-exitcode=1 --leak-check=full \
+              --show-reachable=yes ./icuinfo;
+
+  clang-valgrind-intltest:
+    runs-on: ubuntu-20.04
+    strategy:
+      # "fail-fast: false" lets other jobs keep running even if the test breaks in some other test.
+      fail-fast: false
+      matrix:
+        case: [utility, normalize, collator, regex, format, translit,
+              rbbi, rbnf, rbnfrt, icuserv, idna, convert, rbnfp, csdet,
+              spoof, bidi]
+    steps:
+      - name: Install valgrind
+        run: |
+          set -ex;
+          sudo apt-get -y update;
+          sudo apt-get install -y valgrind;
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build with debug enable
+        env:
+          CC: clang
+          CXX: clang++
+        run: |
+          cd icu4c/source;
+          ./runConfigureICU --enable-debug Linux --disable-renaming && make -j4 tests;
+      - name: Check all top path in intltest is covered
+        run: |
+          cd icu4c/source/test/intltest;
+          if [ $(LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH \
+              ./intltest LIST|egrep "^   [a-z]+$" |wc -l) != "16" ]
+          then
+              echo ICU add more top level test to intltest!
+              echo please update this yml to include those top level test in
+              echo the "case" part.
+              exit -1
+          fi
```

Suggestion to make the check based on the cases rather than just the count of the cases:

```suggestion
          if [ $(LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH \
              ./intltest LIST|egrep "^   [a-z]+$" | awk '{print $1;}' | sort | tr '\n' '|') != "bidi|collate|convert|csdet|format|icuserv|idna|normalize|rbbi|rbnf|rbnfp|rbnfrt|regex|spoof|translit|utility|" ]
          then
              echo ICU has made changes to the top level tests in intltest!
              echo Please update this workflow file to include those top level tests in
              echo the "case" list.
              exit -1
          fi
```